src/metabase/lib/js.cljs

```diff
@@ -418,6 +421,67 @@
   [a-query stage-number column]
   (lib.core/remove-field a-query stage-number column))
 
+(def ^:private visible-columns-defaults
+  {:include-joined?              true
+   :include-expressions?         true
+   :include-implicitly-joinable? true})
+
+
+;; TODO: Added as an expedient to fix metabase/metabase#32373. Due to the interaction with viz-settings, this issue
+;; was difficult to fix entirely within MLv2. Once viz-settings are ported, this function should not be needed, and the
+;; FE logic using it should be ported to MLv2 behind more meaningful names.
+(defn ^:export visible-columns
+  "Return a sequence of column metadatas for columns visible at the given stage of the query. Takes a map of options
+  which defines the set of columns to return. All the options default to *true*, ie. everything is returned by default.
+  ```
+  {
+    include_joined: true,
+    include_expressions true,
+    include_implicitly_joinable: true,
+  }
+  ```"
+  [a-query stage-number ^js js-options]
+  (let [options     (merge visible-columns-defaults
+                           (when-let [opts (some-> js-options js-keys set)]
+                             (cond-> {}
+                               (opts "includeJoined")
+                               (assoc :include-joined? (.-includeJoined js-options))
+                               (opts "includeExpressions")
+                               (assoc :include-expressions? (.-includeExpressions? js-options))
+                               (opts "includeImplicitlyJoinable")
+                               (assoc :include-implicitly-joinable? (.-includeImplicitlyJoinable js-options)))))
```

Comment:
```suggestion
  (let [options     (cond-> visible-columns-defaults
                      (gobj/containsKey js-options "includeJoined")
                      (assoc :include-joined? (.-includeJoined js-options))
                      (gobj/containsKey js-options "includeExpressions")
                      (assoc :include-expressions? (.-includeExpressions? js-options))
                      (gobj/containsKey js-options "includeImplicitlyJoinable")
                      (assoc :include-implicitly-joinable? (.-includeImplicitlyJoinable js-options)))
```
Or maybe even just use `(gobj/get js-options "includeJoined" true)` instead of relying on `visible-columns-defaults`.

Reply:
I went with the latter option, that works fairly well.